src/psyclone/psyir/symbols/symbol.py

```diff
@@ -323,5 +415,40 @@ def is_unresolved(self):
         '''
         return isinstance(self._interface, UnresolvedInterface)
 
+    def find_symbol_table(self, node):
```

Comment:
The implementation looks good but looking where it is used I think it will benefit from a more descriptive name. Maybe something like
`find_[belonging|residing_in]_symbol_table` but I am not sure. What do you think?

Reply:
I thing `symbol.find_symbol_table()` is reasonably descriptive already. Maybe `symbol.containing_symbol_table()`? We could open the vote out to @rupertford?

Reply:
I was over-complicating things myself in this instance. After a second read I think the `symbol.find_symbol_table(node)` is probably fine.